pkg/controller/clusters/validations.go

```diff
@@ -39,11 +39,26 @@ func (v *ClusterValidator) ValidateManagementClusterName(ctx context.Context, lo
 		Namespace: cluster.Namespace,
 		Name:      cluster.Spec.ManagementCluster.Name,
 	}
+	// If we are unable to find the management cluster using the same namespace as the current cluster, we will attempt
+	// to get a list of the clusters with that name across all the namespaces. If we find multiple, which usually should
+	// not happen as these clusters get mapped to a cluster-api cluster object in the eksa-system namespace, then we
+	// also error on that because it is not possible to have multiple resources with the same name within a namespace.
 	if err := v.client.Get(ctx, mgmtClusterKey, mgmtCluster); err != nil {
 		if apierrors.IsNotFound(err) {
```

Comment:
nit: to avoid this level of nesting, you can have a guard like `if err != nil && !apierrors.IsNotFound(err) return err`